src/qibochem/driver/molecule.py

```diff
@@ -182,12 +190,34 @@ def run_pyscf(self, max_scf_cycles=50):
         self.ca = np.asarray(pyscf_job.mo_coeff)  # MO coeffcients
         self.norb = self.ca.shape[1]
         self.nso = 2 * self.norb
+        self.fa = pyscf_job.get_fock()
         self.overlap = np.asarray(pyscf_mol.intor("int1e_ovlp"))
 
         # Currently unused properties?
         # self.ja = pyscf_job.get_j()
         # self.ka = pyscf_job.get_k()
-        # self.fa = pyscf_job.get_fock()
+
+        if do_mp2:
+
+            mp2 = pyscf.mp.MP2(pyscf_job)
+            self.mp2_E, self.mp2_t2 = mp2.kernel(pyscf_job.mo_energy, pyscf_job.mo_coeff)
+            self.mp2_rdm1 = mp2.make_rdm1()  # One body density matrix
+            mp2_virtual_no_occ, mp2_virtual_no = scipy.linalg.eigh(self.mp2_rdm1[self.nalpha :, self.nalpha :])
+            self.mp2_virtual_no_occ = mp2_virtual_no_occ[::-1]
+            self.mp2_virtual_no = mp2_virtual_no[:, ::-1]
+            # Cast the natural orbitals in AO basis
+            v_ao = self.ca[:, self.nalpha :] @ self.mp2_virtual_no.transpose()
```

Comment:
```suggestion
            v_ao = self.ca[:, self.nalpha :] @ self.mp2_virtual_no
```

The rest of the code seems OK to me except for this line, I think you shouldn't need the transpose here?